### PR TITLE
feat: Use custom pre-commit config file

### PR DESCRIPTION
**Pull request by tdurk93**
secureli-373

Migrates `.pre-commit-config.yaml` into the `.secureli/` directory. This directory was previously ignored by git, so I moved its existing contents into `.secureli/local/` and told git to only ignore that subfolder. Using a nonstandard path for `.pre-commit-config.yaml` will allow our tool to not destroy any existing configuration/hooks, and allows us to sidestep much of the complexity of attempting to merge our hooks with existing hooks.

There are a lot of repeated hardcoded strings in this PR. I'm inclined to not do too much refactoring/rework to clean that up, since I'd like to do some significant refactoring in the near future.

## Open Questions

* I opted to keep the file name the same, but I don't love having a hidden file in the hidden folder (feels excessively "hidden"). I'm open to suggestions of whether `pre-commit-config.yaml` is better than `.pre-commit-config.yaml`.
* This will break existing installations of secureli, so we should probably think about adding an automatic migration path prior to merging this. When `secureli` is invoked in a repo with an existing `secureli` installation, the user is met with this prompt: `seCureLI has not yet been installed, install now? [Y/n]:`

## Testing

* Existing unit tests pass
* Manually tested by running `secureli init` in a new repository, then validated the contents & paths of the generated files.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -244,5 +244,5 @@ fabric.properties
 *.drawio.dtmp
 
 # Secureli-generated files (do not modify):
-.secureli
+.secureli/local
```
Comment (kevin-orlando) [inline .gitignore:247]: You will need `.secureli/logs` here as well.

Comment (tdurk93) [reply .gitignore:247]: I moved the logs folder into `local` (for new installations anyway). But I think I'm actually going to revert some of this work and keep the original locations to simplify the migration/reduce breaking changes.

**Comment by tdurk93** [comment]
This PR still needs some unit tests but otherwise I think it's pretty much ready for review.

**Review by isaac-heist-slalom — APPROVED** [review]
Looks good!

**Pull request merged**